wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Scrolling.swift

```diff
@@ -52,11 +52,22 @@ extension ConversationContentViewController {
         updateTableViewHeaderView()
     }
 
-    func scrollToBottom() {
-        guard !isScrolledToBottom else { return }
+    /// Scrolls the tableView to the bottom-most row.
+    ///
+    /// This method checks if the tableView is not already scrolled to the bottom.
+    func scrollToBottomIfNeeded() {
+        let maxMessageCountForAnimation = 20
```

Comment:
Why do you think we should have `maxMessageCountForAnimation` ?

Reply:
The reason behind this is performance. What if we have to scroll through hundreds or more messages. It would degrade the performance. So that's why we have a limit of 20. Over 20 -> we don't animate. Less than 20 we animate.

Reply:
Doesn't iOS optimize this though? What happens when you tap on the top status bar while at the beginning of a long conversation? It should scroll to the bottom with an animation.

Reply:
I don't know how they optimize though and if they're using any private APIs. I was checking the rest of the messenger apps and they're doing what I'm trying to do here regarding animation based on the messages.